packages/dev/core/src/Meshes/Node/Blocks/linearClonerBlock.ts

```diff
@@ -0,0 +1,264 @@
+import { NodeGeometryBlock } from "../nodeGeometryBlock";
+import type { NodeGeometryConnectionPoint } from "../nodeGeometryBlockConnectionPoint";
+import { RegisterClass } from "../../../Misc/typeStore";
+import { NodeGeometryBlockConnectionPointTypes } from "../Enums/nodeGeometryConnectionPointTypes";
+import { Quaternion, Vector3 } from "../../../Maths/math.vector";
+import { type NodeGeometryBuildState } from "../nodeGeometryBuildState";
+import { VertexData } from "core/Meshes/mesh.vertexData";
+
+/**
+ * Block used to clone geometry along a line
+ */
+export class LinearClonerBlock extends NodeGeometryBlock {
+    private _vertexData: VertexData;
+    private _currentIndex: number;
+
+    /**
+     * Gets the current index in the current flow
+     * @returns the current index
+     */
+    public getExecutionIndex(): number {
+        return this._currentIndex;
+    }
+
+    /**
+     * Gets the current face index in the current flow
+     * @returns the current face index
+     */
+    public getExecutionFaceIndex(): number {
+        return 0;
+    }
+
+    /**
+     * Create a new LinearClonerBlock
+     * @param name defines the block name
+     */
+    public constructor(name: string) {
+        super(name);
+        this.registerInput("geometry", NodeGeometryBlockConnectionPointTypes.Geometry, false);
+        this.registerInput("count", NodeGeometryBlockConnectionPointTypes.Int, true, 1, 0);
+        this.registerInput("origin", NodeGeometryBlockConnectionPointTypes.Vector3, true, new Vector3(0, 0, 0));
+
+        //Direction is magnitude per step, or total distance
+        this.registerInput("direction", NodeGeometryBlockConnectionPointTypes.Vector3, true, new Vector3(1, 0, 0));

```

Comment:
To simplify it we should simply do all the transformations per step
If the user wants to add a final transformation they can use a transform node

Reply:
Per step and total is for sure a needed thing for a cloner.

I guess they can figure out their final value and divide it to use the perStep in the same way, but that creates quite a bit of extra work for the designer.

Reply:
Yeah I see it now...the final matrix is hard to find..you are right